lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -3969,6 +3970,11 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
     return BM->addExtInst(Ty, BM->getExtInstSetId(SPIRVEIS_OpenCL), ExtOp,
                           std::move(Operands), BB);
   }
+  case Intrinsic::uadd_with_overflow: {
```

Comment:
Normal transType will not work for result here.
For example: if we have LLVM result type {i16, i1}, we are going to get {TypeInt, TypeBool} for SPIR-V. Instead, we need {TypeInt, TypeInt} according to spec: https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpIAddCarry

The test is also failing in CI (spirv-val) because of this. Please take a look.

Thanks

Reply:
Done